src/allmydata/storage_client.py

```diff
@@ -54,6 +54,24 @@
 # look like?
 #  don't pass signatures: only pass validated blessed-objects
 
+import os
+
+
+def sort_servers(servers, key):
+    """
+    This is sample server selection hookup function.
+    This is the fallback for no 3rd party server selection configured in tahoe.cfg also.
+    The 3rd party can provide a similar server selection function.
+
+    :param servers: A set of allmydata.storage_client.NativeStorageServer object
+    :param key: Specifies a function of one argument that is used to extract a comparison key from Tahoe server list
+                element. The 3rd party can use this as a default comparison key if they don't want to provide their own
+                comparison algorithm but still want to keep using the server selection framework.
+    :return: A sorted server list
```

Comment:
We discussed this in a Tahoe Nuts and Bolts meeting, and decided that this should be passed the ``peer_selection_index``. If we don't pass that, then the hook does not have enough information to make changes to how the ``_permuted`` algorithm [below](https://github.com/tahoe-lafs/tahoe-lafs/pull/140/files#diff-ee280917ba0c482c38e1df0715c41d05R143) works.